### PR TITLE
SU_LOGIN function can receive target user

### PR DESCRIPTION
**Pull request by bashu**
If the SU_LOGIN function takes two arguments, the target
user will be passed, if available, as the second argument.

If the target user is not available, the second argument
will be None.

### CODE REVIEW

File: src/django_su/views.py [hunk 1]
```diff
@@ -22,8 +34,11 @@ def login_as_user(request, user_id):
     return HttpResponseRedirect(getattr(settings, "SU_REDIRECT_LOGIN", "/"))
 
 
-@user_passes_test(can_su_login)
+@login_required
 def su_login(request, user_form=UserSuForm):
+    su_user = get_object_or_404(User, pk=user_id)
```
Comment (betonimig) [inline src/django_su/views.py]: Error! user_id is not defined

File: src/django_su/views.py [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
     su_user = get_object_or_404(User, pk=user_id)
+    if not can_su_login(request.user, su_user):
+        # Following is copied from the user_passes_test source
+        path = request.build_absolute_uri()
+        login_scheme, login_netloc = urlparse.urlparse(settings.LOGIN_URL)[:2]
+        current_scheme, current_netloc = urlparse.urlparse(path)[:2]
+        if ((not login_scheme or login_scheme == current_scheme) and
+            (not login_netloc or login_netloc == current_netloc)):
+            path = request.get_full_path()
+        return redirect_to_login(path)
```
Comment (betonimig) [inline src/django_su/views.py]: rewrite
```
+       # Following is copied from the user_passes_test source
+        path = request.build_absolute_uri()
+        login_scheme, login_netloc = urlparse.urlparse(settings.LOGIN_URL)[:2]
+        current_scheme, current_netloc = urlparse.urlparse(path)[:2]
+        if ((not login_scheme or login_scheme == current_scheme) and
+            (not login_netloc or login_netloc == current_netloc)):
+            path = request.get_full_path()
+        return redirect_to_login(path)
```
to
```
from django.shortcuts import redirect
return redirect(request.META.get("HTTP_REFERER", "/"))```

**Comment by betonimig** [comment]
this pull request is incorect!

**Comment by bashu** [comment]
where?

**Comment by betonimig** [comment]
see comments above

**Comment by bashu** [comment]
thanks!

**Comment by adamcharnock** [comment]
Hey @bashu. Do you know what the status of this PR is?

Oh wow, I've just seen that it is **old**. I'm going to close but feel free to reopen if it is still relevant.

**Pull request closed without merging**